tests/flow_validation/test_choice_state.py

```diff
@@ -0,0 +1,429 @@
+    },
+}
+
+extra_fields_set_on_choice_state = {
+    "StartAt": "ChoiceState",
+    "States": {
+        "ChoiceState": {
+            "Type": "Choice",
+            "Comment": "No info",
+            "ThisIsAnExtraField": True,
+            "Choices": [
+                {
+                    "Variable": "$.SomePath.status",
+                    "StringEquals": "FAILED",
+                    "Next": "ChoiceAState",
+                },
+                {
+                    "Variable": "$.SomePath.status",
+                    "StringEquals": "FAILED",
+                    "Next": "DefaultState",
+                },
+            ],
+            "Default": "DefaultState",
+        },
+        "DefaultState": {
+            "Type": "Fail",
+        },
+        "ChoiceAState": {
+            "Type": "Fail",
+        },
+    },
+}
+
+choice_state_using_or_where_next_is_defined = {
+    "StartAt": "ChoiceState",
+    "States": {
+        "ChoiceState": {
+            "Type": "Choice",
+            "Comment": "No info",
+            "Choices": [
+                {
+                    "Or": [
+                        {
+                            "Variable": "$.SomePath.status",
+                            "StringEquals": "FAILED",
+                            "Next": "ChoiceAState",
+                        },
+                        {
+                            "Variable": "$.SomePath.status",
+                            "StringEquals": "FAILED",
+                        },
+                    ],
+                    "Next": "ChoiceAState",
+                }
+            ],
+            "Default": "DefaultState",
+        },
+        "DefaultState": {
+            "Type": "Fail",
+        },
+        "ChoiceAState": {
+            "Type": "Fail",
+        },
+    },
+}
+
+choice_state_using_not_where_next_is_defined = {
+    "StartAt": "ChoiceState",
+    "States": {
+        "ChoiceState": {
+            "Type": "Choice",
+            "Comment": "No info",
+            "Choices": [
+                {
+                    "Not": {
+                        "Variable": "$.SomePath.status",
+                        "StringEquals": "FAILED",
+                        "Next": "ChoiceAState",
+                    },
+                    "Next": "ChoiceAState",
+                }
+            ],
+            "Default": "DefaultState",
+        },
+        "DefaultState": {
+            "Type": "Fail",
+        },
+        "ChoiceAState": {
+            "Type": "Fail",
+        },
+    },
+}
+
+choice_state_with_invalid_comparison_operator = {
+    "StartAt": "ChoiceState",
+    "States": {
+        "ChoiceState": {
+            "Type": "Choice",
+            "Comment": "No info",
+            "Choices": [
+                {
+                    "Variable": "$.SomePath.status",
+                    "NOT_AN_OPERATOR": "FAILED",
+                    "Next": "ChoiceAState",
+                },
+            ],
+            "Default": "DefaultState",
```

Comment:
All of these test cases have a `"Default"` key. It would be good to confirm what happens if no default is specified (should be fine) and if the default state doesn't actually exist..

Reply:
Good idea for tests! Added tests for both those cases.